libgit/browser.go

```diff
@@ -239,13 +260,18 @@ func (b *Browser) ReadDir(p string) (fis []os.FileInfo, err error) {
 		}
 	}
 
+	childrenPathsToCache := []string(nil)
 	for _, e := range dirTree.Entries {
 		fi, err := b.Lstat(e.Name)
 		if err != nil {
 			return nil, err
 		}
 		fis = append(fis, fi)
+		childrenPathsToCache = append(childrenPathsToCache, path.Join(b.root, e.Name))
```

Comment:
Should this be `path.Join(cachePath, e.Name)`?

Reply:
I think it should be joined against b.root. The b.Lstat above is relative to the root, so e.Name must already include the directory it's in right?

Reply:
You might be thinking of `os.File.Name()`, which is the whole path passed to `Open`?  For an `os.FileInfo.Name` (https://golang.org/pkg/os/#FileInfo), it's described as just the "base name".  And that's what `libkbfs.Node.GetBasename()` returns as well (which is what's used to construct the file info returned by `Lstat()`.

Reply:
As talked offline, `b.Lstat` is supposed to take `path.Join(p, e.Name)` too. Fixed!